src/GraphQL/GraphQLEventSource.cs

```diff
@@ -0,0 +1,97 @@
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+namespace GraphQL;
+
+/// <summary>
+/// EventSource events emitted from the project.
+/// </summary>
+[EventSource(Name = "GraphQL-NET")]
+internal sealed class GraphQLEventSource : EventSource
+{
+    public static GraphQLEventSource Log { get; } = new();
+
+#if DEBUG
+    public GraphQLEventSource()
+        : base(EventSourceSettings.ThrowOnEventWriteErrors)
+    {
+    }
+#endif
+
+    // https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-instrumentation#setting-event-keywords
+#pragma warning disable IDE1006 // Naming Styles
+    public class Keywords
+    {
+        public const EventKeywords Schema = (EventKeywords)1;
+        public const EventKeywords Parsing = (EventKeywords)2;
+        public const EventKeywords Validation = (EventKeywords)4;
+        public const EventKeywords Execution = (EventKeywords)8;
+        public const EventKeywords Introspection = (EventKeywords)16;
+        public const EventKeywords DataLoader = (EventKeywords)32;
+    }
+#pragma warning restore IDE1006 // Naming Styles
+
+    [Event(1, Level = EventLevel.Verbose)]
+    public void RequestIsFilteredOut()
+    {
+        WriteEvent(1);
+    }
+
+    [NonEvent]
+    public void RequestFilterException(Exception ex)
+    {
+        if (IsEnabled(EventLevel.Error, EventKeywords.All))
+        {
+            RequestFilterException(ex.ToInvariantString());
```

Comment:
> Call [IsEnabled()](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource.isenabled#system-diagnostics-tracing-eventsource-isenabled) before performing any resource intensive work related to firing an event, such as computing an expensive event argument that won't be needed if the event is disabled.

`ex.ToInvariantString()` is considered as _intensive work_.

Reply:
See `OpenTelemetry.Instrumentation.AspNetCore.Implementation.AspNetCoreInstrumentationEventSource`